### DISCUSSION TITLE
Request for .NET Core Runtime Support for an Executable JavaScript Script Host

### DISCUSSION THREAD

**Discussion by DevScholar** [dotnet/runtime#87731] [General] [answered]
As a developer, I believe that having a modern, cross-platform JavaScript script host is an important consideration for .NET Core Runtime. While it is true that both macOS and GNOME platforms already have their own JavaScript script hosts, it would be beneficial for Microsoft and .NET Core platform to have one as well.

Existing solutions have several issues. Node.js is difficult to combine with the powerful .NET API, React Native requires packaging before use and relies heavily on GUI and React, and PowerShell has strange syntax that is not suitable for script programming. Windows Script Host, on the other hand, is outdated and not cross-platform.

Combining the simplicity of JavaScript with the strong capabilities of .NET platform would be in line with Microsoft's strategy and beneficial for JavaScript developers. Microsoft already has a library called ClearScript that can achieve this, but a script host that is ready to use and preferably built-in to .NET platform is needed.

Developers can use this script host to write console programs, or they can utilize the powerful capabilities of .NET platform and Webview2 (and Webview2 equivalents for non-Windows platforms), developers can write desktop applications using JavaScript, or use this script host to write desktop applications that use other UI frameworks. Therefore, I request that .NET Core Runtime adds support for a JavaScript script host.

We want something similar to the Windows Script Host, but modern, cross-platform, and based on the .NET platform, and it should be built-in to the .NET runtime.

**Comment by En3Tho** [comment]
`PowerShell has strange syntax that is not suitable for script programming. `

This is a very weird statement. You might personally not like it, but it is certainly suitable for script programming.

As for the script host itself it should have nothing to do with .Net runtime itself because ~~JS is a blight and should be destroyed.~~ you can use a library for this: https://github.com/sebastienros/jint

**Reply by En3Tho** [reply, marked as answer]
https://github.com/microsoft/node-api-dotnet

**Comment by Wraith2** [comment]
> I believe that having a modern, cross-platform JavaScript script host is an important consideration for .NET Core Runtime.

Why? What use cases does it enable having one as part of the runtime that can't be satisfied by a third party solution? If the existing solutions aren't good enough can you not improve them or produce your own version or fork that removes those problems?
